src/translate.rs

```diff
@@ -97,11 +97,45 @@ pub fn header_items(ctxt: &rustast::ExtCtxt) -> Vec<rustast::P<rustast::Item>> {
 	).unwrap());
 
 	items.push(quote_item!(ctxt,
+		#[derive(PartialEq, Eq, Debug)]
+		pub struct ParseError {
+			pub line: usize,
+			pub column: usize,
+			pub offset: usize,
+			pub expected: ::std::collections::HashSet<&'static str>,
+		}
+	).unwrap());
+	
+	items.push(quote_item!(ctxt,
+		pub type ParseResult<T> = Result<T, ParseError>;
+	).unwrap());
+
```

Comment:
We should probably `impl ::std::error::Error for ParseError`, though `description` returns `&str`, so it would have to return something generic like "parse error", and I'm not sure what that trait is actually used for anyway....